analyses/SC_cd_2000/01_prep_2000_SC_cd_2000.R

```diff
@@ -0,0 +1,138 @@
+        join_vtd_shapefile(year = 2000) %>%
+        st_transform(EPSG$SC)
+
+    sc_shp <- sc_shp %>%
+        rename(muni = place) %>%
+        mutate(county_muni = if_else(is.na(muni), county, str_c(county, muni))) %>%
+        relocate(muni, county_muni, cd_1990, .after = county)
+
+    # Create perimeters in case shapes are simplified
+    redistmetrics::prep_perims(shp = sc_shp,
+        perim_path = here(perim_path)) %>%
+        invisible()
+
+    # simplifies geometry for faster processing, plotting, and smaller shapefiles
+    if (requireNamespace("rmapshaper", quietly = TRUE)) {
+        sc_shp <- rmapshaper::ms_simplify(sc_shp, keep = 0.05,
+            keep_shapes = TRUE) %>%
+            suppressWarnings()
+    }
+
+    # create adjacency graph
+    sc_shp$adj <- redist.adjacency(sc_shp)
+
+    sc_shp <- sc_shp %>%
+        fix_geo_assignment(muni)
+
+    write_rds(sc_shp, here(shp_path), compress = "gz")
+    cli_process_done()
+} else {
+    sc_shp <- read_rds(here(shp_path))
+    cli_alert_success("Loaded {.strong SC} shapefile")
+}
+
+
+###############################################################################
+# Logit-shift ndv/nrv to match 2000 MEDSL county results
+###############################################################################
+
+#' Logit Shift Baseline Data
+#'
+#' @param d_baseline baseline data containing vote columns
+#' @param ndv Unquoted Democratic vote column name
+#' @param nrv Unquoted Republican vote column name
+#' @param target target to logit shift to
+#' @param tol
+#'
+#' @returns a data frame with adjusted vote columns
+#' @export
+#'
+#' @examples
+#' # TODO
+logit_shift_baseline <- function(d_baseline, ndv, nrv,
+                                 target = 0.5,
+                                 tol = sqrt(.Machine$double.eps)) {
+    if (missing(ndv) || missing(nrv)) {
+        cli::cli_abort("Both {.arg ndv} and {.arg nrv} must be provided.")
+    }
+    ndv_q <- rlang::enquo(ndv)
+    nrv_q <- rlang::enquo(nrv)
+
+    ndv_vec <- dplyr::pull(d_baseline, !!ndv_q)
+    nrv_vec <- dplyr::pull(d_baseline, !!nrv_q)
+
+    turn <- ndv_vec + nrv_vec
+    ldvs <- dplyr::if_else(turn > 0, log(ndv_vec) - log(nrv_vec), 0)
+
+    res <- uniroot(function(shift) {
+        stats::weighted.mean(plogis(ldvs + shift), turn) - target
+    }, c(-1, 1), tol = tol)
+
+    ldvs <- ldvs + res$root
+
+    ndv_new <- turn*plogis(ldvs)
+    nrv_new <- turn - ndv_new
+
+    dplyr::mutate(
+        d_baseline,
+        !!rlang::as_name(ndv_q) := ndv_new,
+        !!rlang::as_name(nrv_q) := nrv_new
+    )
+}
+
+# 1. Load the MEDSL county CSV as `medsl_cty` ----
+medsl_cty <- read_csv(
+    here("data-raw/baseline_voteshare_medsl_00.csv"),
+    show_col_types = FALSE
+)
+
+# 2. Add county_fips column based on VTD GEOID ----
+sc_shp <- sc_shp |>
+    mutate(county_fips = stringr::str_sub(GEOID, 1, 5))
+
+names(sc_shp)
+
+# 3. For each county, logit-shift ndv/nrv to the 2000 target from MEDSL ----
+sc_shp |>
```

Comment:
I guess a good question here is if we should be running this separately in a file 04_? We have done this in specific cases (often TX and FL where we have longer validations).